src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,82 @@
+// <copyright file="PrometheusExporterEndpointRouteBuilderExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NETCOREAPP3_1_OR_GREATER
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Exporter.Prometheus;
+using OpenTelemetry.Metrics;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="IEndpointRouteBuilder"/> to add
+    /// Prometheus scraping endpoint.
+    /// </summary>
+    public static class PrometheusExporterEndpointRouteBuilderExtensions
```

Comment:
To be consistent with the "Use" counterparts should we add a couple overloads for the basic use cases? I'm thinking...

```csharp
        public static IEndpointConventionBuilder MapPrometheusScrapingEndpoint(this IEndpointRouteBuilder endpoints)
                => this.MapPrometheusScrapingEndpoint(path: null, meterProvider: null, configureBranchedPipeline: null);

        public static IEndpointConventionBuilder MapPrometheusScrapingEndpoint(this IEndpointRouteBuilder endpoints, string path)
                => this.MapPrometheusScrapingEndpoint(path, meterProvider: null, configureBranchedPipeline: null);
```

Reply:
Done.

Though I haven't added a `Guard.ThrowIfNull(path);` statement in there like the other methods. Do you think this is useful?

Reply:
My feeling is the guard on this one is useful:

```csharp
MapPrometheusScrapingEndpoint(this IEndpointRouteBuilder endpoints, string path)
```

Because the user has explicitly called the overload to set the path, but without a path. So we throw to notify them they made a mistake. If they really don't want to pass the path, there are the other overloads to do that.

Reply:
I've updated the overload to include to guard. Please let me know if there is anything else.